src/huggingface_hub/utils/tags.py

```diff
@@ -0,0 +1,140 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Tagging utilities. """
+
+
+class AttributeDictionary(dict):
+    """
+    `dict` subclass that also provides access to keys as attributes
+
+    If a key starts with a number, it will exist in the dictionary
+    but not as an attribute
+
+    Example usage:
+
+        >>> d = AttributeDictionary()
+        >>> d["test"] = "a"
+        >>> print(d.test) # prints "a"
+
+    """
+
+    def __getattr__(self, k):
+        if k in self:
+            return self[k]
+        else:
+            raise AttributeError(k)
+
+    def __setattr__(self, k, v):
+        (self.__setitem__, super().__setattr__)[k[0] == "_"](k, v)
+
+    def __delattr__(self, k):
+        if k in self:
+            del self[k]
+        else:
+            raise AttributeError(k)
+
+    def __dir__(self):
+        keys = list(self.keys())
+        keys = [key for key in keys if not key[0].isdigit()]
```

Comment:
This leads to an edge case with size tags having a digit as first char

Reply:
Yeah, this was an initial hacky-fix. Need to think on it more. Essentially the idea behind this is while it doesn't show up in `dir`, it *will* show up in the `__repr__`, and there are comments to direct the user to do `["123"]` since that's a python limitation.

IIUC a similar thing would happen here, no? Since we can't have any attributes as digits in python

Reply:
*Ah wait*. Strike that. It's indeed a bug. Will work towards a fix

Reply:
@osanseviero proposed solution:
Here is how they'd be displayed (push coming momentarily):
Available Attributes or Keys:
 * 10K<n<100K (Key only)
 * 1K<n<10K (Key only)
 * 100K<n<1M (Key only)
 * n<1K
 * 1M<n<10M (Key only)
 * unknown
 * 10M<n<100M (Key only)
 * 100M<n<1B (Key only)
 * 1B<n<10B (Key only)
 * 10K<n<100k (Key only)
 * n>1M
 * 2G<n<1B (Key only)
 * 90K<n<150K (Key only)
 * 1K (Key only)
 * 18K (Key only)
 * 8K (Key only)
 * 10k<n<100k (Key only)
 * 54KB (Key only)
 * 8k<n<10k (Key only)